Source/Core/DolphinQt/EmulatedUSB/WiiSpeakWindow.cpp

```diff
@@ -0,0 +1,149 @@
+  checkbox_mic_muted->setChecked(Config::Get(Config::MAIN_WII_SPEAK_MUTED));
+  connect(checkbox_mic_muted, &QCheckBox::toggled, this,
+          &WiiSpeakWindow::SetWiiSpeakConnectionState);
+  checkbox_mic_muted->setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Minimum);
+  config_layout->addWidget(checkbox_mic_muted);
+
+  auto* volume_layout = new QGridLayout();
+  static constexpr int FILTER_MIN = -50;
+  static constexpr int FILTER_MAX = 50;
+  const int volume_modifier =
+      std::clamp<int>(Config::Get(Config::MAIN_WII_SPEAK_VOLUME_MODIFIER), FILTER_MIN, FILTER_MAX);
+  auto filter_slider = new QSlider(Qt::Horizontal, this);
+  auto slider_label = new QLabel(tr("Volume modifier (value: %1dB)").arg(volume_modifier));
+  connect(filter_slider, &QSlider::valueChanged, this, [slider_label](int value) {
+    Config::SetBaseOrCurrent(Config::MAIN_WII_SPEAK_VOLUME_MODIFIER, value);
+    slider_label->setText(tr("Volume modifier (value: %1dB)").arg(value));
+  });
+  filter_slider->setMinimum(FILTER_MIN);
+  filter_slider->setMaximum(FILTER_MAX);
+  filter_slider->setValue(volume_modifier);
+  filter_slider->setTickPosition(QSlider::TicksBothSides);
+  filter_slider->setTickInterval(10);
+  filter_slider->setSingleStep(1);
+  volume_layout->addWidget(new QLabel(QStringLiteral("%1dB").arg(FILTER_MIN)), 0, 0, Qt::AlignLeft);
+  volume_layout->addWidget(slider_label, 0, 1, Qt::AlignCenter);
+  volume_layout->addWidget(new QLabel(QStringLiteral("%1dB").arg(FILTER_MAX)), 0, 2,
+                           Qt::AlignRight);
+  volume_layout->addWidget(filter_slider, 1, 0, 1, 3);
+  config_layout->addLayout(volume_layout);
+  config_layout->setStretch(1, 3);
+
+  m_combobox_microphones = new QComboBox();
+#ifndef HAVE_CUBEB
+  m_combobox_microphones->addItem(QLatin1String("(%1)").arg(tr("Audio backend unsupported")),
+                                  QString{});
+#else
+  m_combobox_microphones->addItem(QLatin1String("(%1)").arg(tr("Autodetect preferred microphone")),
+                                  QString{});
+  for (auto& [device_id, device_name] : CubebUtils::ListInputDevices())
+  {
+    const auto user_data = QString::fromStdString(device_id);
+    m_combobox_microphones->addItem(QString::fromStdString(device_name), user_data);
+  }
+#endif
+  connect(m_combobox_microphones, &QComboBox::currentIndexChanged, this,
+          &WiiSpeakWindow::OnInputDeviceChange);
+
+  auto current_device_id = QString::fromStdString(Config::Get(Config::MAIN_WII_SPEAK_MICROPHONE));
+  m_combobox_microphones->setCurrentIndex(m_combobox_microphones->findData(current_device_id));
+  config_layout->addWidget(m_combobox_microphones);
+
+  config_group->setLayout(config_layout);
+  main_layout->addWidget(config_group);
+
+  setLayout(main_layout);
+}
+
+void WiiSpeakWindow::EmulateWiiSpeak(bool emulate)
+{
+  Config::SetBaseOrCurrent(Config::MAIN_EMULATE_WII_SPEAK, emulate);
+}
+
+void WiiSpeakWindow::SetWiiSpeakConnectionState(bool muted)
```

Comment:
The function is called `SetWiiSpeakConnectionState` but the setting is called `MAIN_WII_SPEAK_MUTED`?

Reply:
This was an oversight when renaming the config from MAIN_WII_SPEAK_CONNECTED to MAIN_WII_SPEAK_MUTED. Should be fixed, now.